src/js/components/Navigation/SettingsPersonalSideBar.jsx

```diff
@@ -70,10 +70,41 @@ export default class SettingsPersonalSideBar extends Component {
       isOnPartnerUrl: AppObservableStore.isOnPartnerUrl(),
     });
   }
-
+// helper function 
+  fireSettingsGTMEvent = ({ buttonId, destinationPath = '', actionType = 'navigate' }) => {
+    TagManager.dataLayer({
+      dataLayer: {
+        event: 'action',
+        actionDetails: {
+          actionType,
+          buttonId,
+        },
+        ...(destinationPath && {
+          destinationDetails: {
+            destinationPath,
+          },
+        }),
+        pageDetails: {
+          pageTitle: document.title,
+          pagePath: window.location.pathname,
+        },
+        userDetails: {
+          isSignedIn: VoterStore.getVoterIsSignedIn(),
```

Comment:
I don't think you can call the stores to get these variables (VoterIsSignedIn and VoterId) from within you function that is called during render.  React render requires static data.  
So one possible solution would be to put these two variables into state (setState) maybe when componentDidMount is called, and then pass them into your fireSettingsGTMEvent from the call in the render.